### PR TITLE
Add B3 multi header propagation support

### PR DESCRIPTION
**Pull request by seemk**
Original PR: https://github.com/open-telemetry/opentelemetry-cpp-contrib/pull/103

I've brought it up to date and added tests, retained the original commits by @kuzaxak

### CODE REVIEW

File: instrumentation/nginx/src/otel_ngx_module.cpp [hunk 1]
```diff
@@ -224,6 +235,46 @@ nostd::string_view WithoutOtelVarPrefix(ngx_str_t value) {
...
+
+  if (!IsOtelEnabled(req)) {
+    v->valid = 0;
+    v->not_found = 1;
+    return NGX_OK;
+  }
+
+  TraceContext* traceContext = GetTraceContext(req);
+
+  if (traceContext == nullptr || !traceContext->request_span) {
```
Comment (lalitb) [inline instrumentation/nginx/src/otel_ngx_module.cpp:250]:
```suggestion
  if (!traceContext || !traceContext->request_span) {
```

doing `traceContext->request_span` is risky if `traceContext` is nullptr.

**Review by lalitb — APPROVED** [review]
Thanks

File: instrumentation/nginx/src/otel_ngx_module.cpp (hunk 1 above)
Comment (seemk) [reply instrumentation/nginx/src/otel_ngx_module.cpp:250]: These lines are equal (`traceContext == nullptr` and `!traceContext`) so a null `traceContext` isn't accessed here

File: instrumentation/nginx/src/otel_ngx_module.cpp [hunk 2]
```diff
@@ -224,6 +235,46 @@ nostd::string_view WithoutOtelVarPrefix(ngx_str_t value) {
...
+    ngx_log_error(
+        NGX_LOG_ERR, req->connection->log, 0,
+        "Unable to get trace context when getting span id");
```
[the remark this answers is not in the document]
Comment (seemk) [reply instrumentation/nginx/src/otel_ngx_module.cpp:253]: I've signed the CLA, but for some reason I'm getting an EasyCLA error when pushing to this branch, will change the wording in another PR.

**Pull request merged**